ocean_provider/file_types/definitions.py

```diff
@@ -103,27 +112,25 @@ def check_details(self, with_checksum=False):
 
     def _get_result_from_url(self, with_checksum=False):
         lightweight_methods = [] if self.method == "post" else ["head", "options"]
-
-        for method in lightweight_methods:
-            url = self.get_download_url()
-            func = getattr(requests, method)
-            result = func(
-                url,
-                timeout=REQUEST_TIMEOUT,
-                headers=self.headers,
-                params=self.format_userdata(),
-            )
-
-            if (
-                not with_checksum
-                and result.status_code == 200
-                and (
-                    result.headers.get("Content-Type")
-                    or result.headers.get("Content-Range")
+        if not with_checksum:
```

Comment:
I suggest if with_checksum we should return the other block directly, making the code less imbricated and more readable.